File: diagramas/ChefApp/ChefApp/app/src/main/java/com/chefapp/activities/MainActivity.java
package com.chefapp.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.chefapp.R;
import com.chefapp.fragments.BotonesFragment;
import com.chefapp.fragments.FotosFragment;
import com.chefapp.fragments.PerfilFragment;
import com.chefapp.fragments.VideoFragment;
import com.chefapp.fragments.WebFragment;

public class MainActivity extends AppCompatActivity {

    private LinearLayout navPerfil, navFotos, navVideo, navWeb, navBotones;
    private int currentNav = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        navPerfil  = findViewById(R.id.nav_perfil);
        navFotos   = findViewById(R.id.nav_fotos);
        navVideo   = findViewById(R.id.nav_video);
        navWeb     = findViewById(R.id.nav_web);
        navBotones = findViewById(R.id.nav_botones);

        navPerfil.setOnClickListener(v  -> selectNav(0));
        navFotos.setOnClickListener(v   -> selectNav(1));
        navVideo.setOnClickListener(v   -> selectNav(2));
        navWeb.setOnClickListener(v     -> selectNav(3));
        navBotones.setOnClickListener(v -> selectNav(4));

        // Default: Perfil
        selectNav(0);
    }

    private void selectNav(int index) {
        if (currentNav == index) return;
        currentNav = index;

        updateNavUI();
        loadFragment(index);
    }

    private void updateNavUI() {
        LinearLayout[] navItems = {navPerfil, navFotos, navVideo, navWeb, navBotones};

        for (int i = 0; i < navItems.length; i++) {
            LinearLayout item = navItems[i];
            boolean active = (i == currentNav);

            item.setBackgroundResource(active ? R.drawable.bg_sidebar_active : 0);

            // Tint all children
            for (int j = 0; j < item.getChildCount(); j++) {
                View child = item.getChildAt(j);
                int colorRes = active ? R.color.text_white : R.color.sidebar_icon;
                int color = getResources().getColor(colorRes, null);

                if (child instanceof ImageView) {
                    ((ImageView) child).setColorFilter(color);
                } else if (child instanceof android.widget.TextView) {
                    ((android.widget.TextView) child).setTextColor(color);
                }
            }
        }
    }

    private void loadFragment(int index) {
        Fragment fragment;
        switch (index) {
            case 1:  fragment = new FotosFragment();   break;
            case 2:  fragment = new VideoFragment();   break;
            case 3:  fragment = new WebFragment();     break;
            case 4:  fragment = new BotonesFragment(); break;
            default: fragment = new PerfilFragment();  break;
        }

        FragmentTransaction tx = getSupportFragmentManager().beginTransaction();
        tx.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out);
        tx.replace(R.id.fragment_container, fragment);
        tx.commit();
    }
}

File: app/src/main/java/com/example/chefapp/MainActivity.kt
package com.example.chefapp

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.fragment.app.Fragment
import com.example.chefapp.databinding.ActivityMainBinding
import com.example.chefapp.fragments.*

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var drawerToggle: ActionBarDrawerToggle

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)
        supportActionBar?.title = "🍳 ChefApp"

        configurarDrawer()
        cargarFragmentoIzquierdo()
        cargarFragmentoDerecho(PerfilFragment(), "perfil")    }

    private fun configurarDrawer() {
        drawerToggle = ActionBarDrawerToggle(
            this,
            binding.drawerLayout,
            binding.toolbar,
            R.string.app_name,
            R.string.app_name
        )
        binding.drawerLayout.addDrawerListener(drawerToggle)
        drawerToggle.syncState()

        binding.navigationView.setNavigationItemSelectedListener { menuItem ->
            manejarItemDrawer(menuItem)
            binding.drawerLayout.closeDrawer(GravityCompat.START)
            true
        }
    }

    private fun cargarFragmentoIzquierdo() {
        supportFragmentManager.beginTransaction()
            .replace(R.id.frameLeft, LeftFragment())
            .commit()
    }

    fun cargarFragmentoDerecho(fragment: Fragment, tag: String) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.frameRight, fragment, tag)
            .commit()

        val leftFrag = supportFragmentManager.findFragmentById(R.id.frameLeft) as? LeftFragment
        leftFrag?.setItemActivo(tag)
    }

    private fun manejarItemDrawer(menuItem: MenuItem) {
        when (menuItem.itemId) {
            R.id.nav_perfil   -> cargarFragmentoDerecho(PerfilFragment(), "perfil")
            R.id.nav_fotos    -> cargarFragmentoDerecho(FotosFragment(), "fotos")
            R.id.nav_video    -> cargarFragmentoDerecho(VideoFragment(), "video")
            R.id.nav_web      -> cargarFragmentoDerecho(WebFragment(), "web")
            R.id.nav_botones  -> cargarFragmentoDerecho(BotonesFragment(), "botones")
            R.id.nav_cerrar   -> cerrarSesion()
        }
    }

    private fun cerrarSesion() {
        finish()
    }

    override fun onBackPressed() {
        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
            binding.drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }
}